### PR TITLE
Fresh constants + MakePlot.C creates template plots + minor cosmetic changes

### PR DESCRIPTION
**Pull request by jozzez1**
- This code now also produces the plots from the dEdx template.
- Fresh K and C constants.
- Symmetrized the geometrical cuts for strip modules.
- Fixed the pixel scale factor -- now it is perfect. If I make a control to recomputed the dEdxSF when it has already been applied I get 0.999 for all scale factors

### CODE REVIEW

**Comment by jozzez1** [comment]
- logscale for the ROC curve
- plots related to Charge_Vs_XY*

**Comment by jozzez1** [comment]
Some more goodies in the DeDxStudy. Also I fixed a typo in one of the comments in Analysis_CommonFunction.h.

**Comment by jozzez1** [comment]
- now the ROC curve is displayed for only HSCP
- signal samples are read correctly for run2 from ingrid
- some other minor style fixes

**Comment by jozzez1** [comment]
Added pixel into the mix. Relevant files are:
Analysis_CommonFunction.h, Analysis_Step1_EventLoop.C and DeDxStudy.C.

File: SUSYBSMAnalysis/HSCP/test/AnalysisCode/Analysis_CommonFunction.h [hunk 1]
```diff
@@ -495,12 +495,17 @@ void LoadDeDxCalibration(std::unordered_map<unsigned int,double>& TrackerGains,
 
 
 
-TH3F* loadDeDxTemplate(string path, bool splitByModuleType){
+TH3F* loadDeDxTemplate(string path, bool splitByModuleType, bool useStrip, bool usePixel){
```
Comment (quertenmont) [inline SUSYBSMAnalysis/HSCP/test/AnalysisCode/Analysis_CommonFunction.h]: I don't want a "useStrip" or "usePixel" in dEdxTemplates.
Templates should contains everything always.  
Using pixel or strip probability should be handled when the templates are used

File: SUSYBSMAnalysis/HSCP/test/AnalysisCode/Analysis_CommonFunction.h [hunk 2]
```diff
@@ -508,14 +513,14 @@ TH3F* loadDeDxTemplate(string path, bool splitByModuleType){
       Prob_ChargePath->RebinX(Prob_ChargePath->GetNbinsX());
    }
 
-   for(int i=0;i<=Prob_ChargePath->GetXaxis()->GetNbins()+1;i++){
+   for(int i=1;i<=Prob_ChargePath->GetXaxis()->GetNbins();i++){// do not include under/overflow here
```
Comment (quertenmont) [inline SUSYBSMAnalysis/HSCP/test/AnalysisCode/Analysis_CommonFunction.h]: why the hell do you decide to change this behaviour?
Under and Over flow bins should be considered

File: SUSYBSMAnalysis/HSCP/test/AnalysisCode/Analysis_CommonFunction.h [hunk 3]
```diff
@@ -531,10 +536,10 @@ TH3F* loadDeDxTemplate(string path, bool splitByModuleType){
 
 #include "DataFormats/SiStripDetId/interface/SiStripDetId.h"
 
-const double TkModGeomThickness[] = {1, 0.029000, 0.029000, 0.047000, 0.047000, 0.029000, 0.029000, 0.029000, 0.029000, 0.029000, 0.029000, 0.029000, 0.047000, 0.047000, 0.047000};
-const double TkModGeomLength   [] = {1, 5.844250, 5.844250, 9.306700, 9.306700,  5.542900, 4.408000, 5.533000, 4.258000, 4.408000, 5.533000, 5.758000, 7.363125, 9.204400, 10.235775};
-const double TkModGeomWidthB   [] = {1, 3.072000, 3.072000, 4.684800, 4.684800, 4.579445, 5.502407, 3.158509, 4.286362, 5.502407, 4.049915, 3.561019, 6.002559, 5.235483, 3.574395};
-const double TkModGeomWidthT   [] = {1, 3.072000, 3.072000, 4.684800,  4.684800, 3.095721, 4.322593, 4.049915, 3.146580, 4.322593, 3.158509, 2.898798, 4.824683, 4.177638,  4.398049};
+const double TkModGeomThickness[] = {1, 0.029000, 0.029000, 0.047000, 0.047000, 0.029000, 0.029000, 0.029000, 0.029000, 0.029000, 0.029000, 0.029000, 0.047000, 0.047000, 0.047000, 1.0};
```
Comment (quertenmont) [inline SUSYBSMAnalysis/HSCP/test/AnalysisCode/Analysis_CommonFunction.h]: don't understand why you are adding a field here.
I guess it comes from inclusive... but inclusive has "no" geometry, so this vector shouldn't be changed

File: SUSYBSMAnalysis/HSCP/test/AnalysisCode/Analysis_CommonFunction.h [hunk 4]
```diff
@@ -579,9 +585,9 @@ bool isHitInsideTkModule(const LocalPoint hitPos, const DetId& detid){
 
 
 
-DeDxData* computedEdx(const DeDxHitInfo* dedxHits, double* scaleFactors, TH3* templateHisto, bool usePixel, bool useClusterCleaning, bool reverseProb, bool useTruncated, std::unordered_map<unsigned int,double>* TrackerGains, bool useStrip, bool mustBeInside){
+DeDxData* computedEdx(const DeDxHitInfo* dedxHits, double* scaleFactors, TH3* templateHisto, bool useClusterCleaning, bool reverseProb, bool useTruncated, std::unordered_map<unsigned int,double>* TrackerGains, bool useStrip, bool usePixel, bool mustBeInside){
```
Comment (quertenmont) [inline SUSYBSMAnalysis/HSCP/test/AnalysisCode/Analysis_CommonFunction.h]: The fact that usePixel and useStrip are at different places is for backward compatibility.
Don't change it

File: SUSYBSMAnalysis/HSCP/test/AnalysisCode/Analysis_CommonFunction.h [hunk 5]
```diff
@@ -625,13 +631,15 @@ DeDxData* computedEdx(const DeDxHitInfo* dedxHits, double* scaleFactors, TH3* te
             if(isSatCluster)NSat++;
         }
 
-	double scaleFactor = scaleFactors[0];
-	if (detid.subdetId()<3) scaleFactor *= scaleFactors[1];
+	     double scaleFactor = scaleFactors[0];
```
Comment (quertenmont) [inline SUSYBSMAnalysis/HSCP/test/AnalysisCode/Analysis_CommonFunction.h]: Change into
double scaleFactor = scaleFactors[detid.subdetId()>=3?0:1];

File: SUSYBSMAnalysis/HSCP/test/AnalysisCode/Analysis_Step1_EventLoop.C [hunk 6]
```diff
@@ -1193,8 +1190,8 @@ void Analysis_Step1_EventLoop(char* SavePath)
 
                //Compute dE/dx on the fly
                //computedEdx(dedxHits, Data/MC scaleFactor, templateHistoForDiscriminator, usePixel, useClusterCleaning, reverseProb)
-               DeDxData* dedxSObj = computedEdx(dedxHits, dEdxSF, dEdxTemplates, false, useClusterCleaning, TypeMode==5, false, TrackerGains.size()>0?&TrackerGains:NULL);
-               DeDxData* dedxMObj = computedEdx(dedxHits, dEdxSF, NULL,          false, useClusterCleaning, false      , false, TrackerGains.size()>0?&TrackerGains:NULL);
+               DeDxData* dedxSObj = computedEdx(dedxHits, dEdxSF, dEdxTemplates, useClusterCleaning, TypeMode==5, false, TrackerGains.size()>0?&TrackerGains:NULL);
```
Comment (quertenmont) [inline SUSYBSMAnalysis/HSCP/test/AnalysisCode/Analysis_Step1_EventLoop.C]: Keep backward compatibility

File: SUSYBSMAnalysis/HSCP/test/UsefulScripts/DeDxStudy/DeDxStudy.C [hunk 7]
```diff
@@ -179,7 +182,11 @@ void DeDxStudy(string DIRNAME="COMPILE", string INPUT="dEdx.root", string OUTPUT
    gStyle->SetNdivisions(505,"X");
    TH1::AddDirectory(kTRUE);
 
-   bool isData = INPUT.find("MC")!=string::npos ? false : true;
+   bool isData   = !(INPUT.find("MC")      !=string::npos) ,
```
Comment (quertenmont) [inline SUSYBSMAnalysis/HSCP/test/UsefulScripts/DeDxStudy/DeDxStudy.C]: I told you several times to proceed differently and to call the signal sample per datasetName, so it uses Analysis_Samplte.txt information

File: SUSYBSMAnalysis/HSCP/test/UsefulScripts/DeDxStudy/DeDxStudy.C [hunk 8]
```diff
@@ -192,22 +199,51 @@ void DeDxStudy(string DIRNAME="COMPILE", string INPUT="dEdx.root", string OUTPUT
...
+      isData   = (sample.Type==0);
+      isSignal = (sample.Type==2);
       GetInputFiles(sample, BaseDirectory, FileName, 0);
    }
 
-   TH3F* dEdxTemplates    = NULL;
-   TH3F* dEdxTemplatesInc = NULL;
+//   TH3F* dEdxTemplates    = NULL;
+//   TH3F* dEdxTemplatesInc = NULL;
+   TH3F* dEdxTemplatesPO      = NULL;
```
Comment (quertenmont) [inline SUSYBSMAnalysis/HSCP/test/UsefulScripts/DeDxStudy/DeDxStudy.C]: Way to many templates which all contains the same information.
you must have only 3
SP
SP_Split
SP_Split_In
Realize the amount of memory consumed by these histograms?

**Comment by jozzez1** [comment]
- Well, I do consider them (under/overflow bins), just not in the geometry. If we have 15 possible geometries (14 for strips + 1 for pixel), wouldn't one extra suddenly popping up mean an error?
- Ok, I will change it so the probability map will include all. I had these flags because yesterday I wasn't aware of the 1.0/265 trick and I was afraid of adding pixel into the inclusive.

File: SUSYBSMAnalysis/HSCP/test/UsefulScripts/DeDxStudy/DeDxStudy.C (hunk 7 above)
Comment (jozzez1) [inline SUSYBSMAnalysis/HSCP/test/UsefulScripts/DeDxStudy/DeDxStudy.C]: But the MC files (minBias) and the data files (ZeroBias) we are using are not in the Analysis_Samples.txt. Signal samples are called per datasetName.

**Comment by jozzez1** [comment]
I did as you asked. You should not yet merge, because in the DeDxStudy.C I am loading my local templates, which are not finished yet, but should be finished soon.

**Comment by jozzez1** [comment]
Just committed the changes.

**Pull request merged**